extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -87,12 +81,27 @@ public Object get(ByteBuffer buf, int position)
   //Note that this is not threadsafe and I don't think it needs to be
   private Union getUnion(ByteBuffer buf, int position)
   {
-    Union union = unions.get(position);
-    if (union == null) {
-      Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
-      union = (Union) SetOperation.wrap(mem);
-      unions.put(position, union);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
+    Union union = unionMap != null ? unionMap.get(position) : null;
+    if (union != null) {
+      return union;
+    }
+    return createNewUnion(buf, position, true);
+  }
+
+  private Union createNewUnion(ByteBuffer buf, int position, boolean isWrapped)
+  {
+    NativeMemory nm = getNativeMemory(buf);
+    Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
+    Union union = isWrapped
+                  ? (Union) SetOperation.wrap(mem)
+                  : (Union) SetOperation.builder().initMemory(mem).build(size, Family.UNION);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
```

Comment:
@leventov results were straightforward that is why I didn't explained it.

>This benchmark is doing computeIfAbsent() against Int2ObjectOpenHashMap rather than IdentityHashMap

https://github.com/druid-io/druid/issues/4034 suggests ComputeIfAbsent  should be used for java 8 map and that is why I performed this benchmark on arbitrary map (Int2ObjectOpenHashMap). If not the case please do not force it in other places.
 I also changed your benchmark to use HashMap instead of IdentityHashMap https://gist.github.com/akashdw/68b160620eff9e2d5bfcebeb42f4fd42.

resulted,
ComputeIfAbsent.withComputeIfAbsent     avgt   25  75.080 ± 1.787  us/op
ComputeIfAbsent.withoutComputeIfAbsent  avgt   25  71.482 ± 1.697  us/op




>why you decided to benchmark code, that is not a hot spot

`getUnion()` is a hot spot.

As I don't see any substantial benefits of using `ComputeIfAbsent`, I'm not going to make requested changes.
At the same time I'm happy to make this change if other committers agree on https://github.com/druid-io/druid/issues/4034.